### PR TITLE
remove oauthclient and update operand SA with annotations

### PR DESCRIPTION
**Pull request by Tirumalavasa**
https://github.ibm.com/IBMPrivateCloud/roadmap/issues/57068

### CODE REVIEW

**Review by rashmi43 — APPROVED** [review]

**Comment by rwhundley** [comment]
@Tirumalavasa or @rashmi43 - would you please also fixup the commits that git log noise prior to merging, too? I'd like for our git log to be more readable, and having bullets for things like initial commits and merges in this context gets in the way of what the commit accomplished (removed OAuthClient support, specific behavior in SA's around annotations.)

File: pkg/controller/client/client_controller.go [hunk 1]
```diff
@@ -583,19 +549,8 @@ func (r *ReconcileClient) deleteClient(ctx context.Context, client *oidcv1.Clien
...
-	// Delete OpenShift OAuthClient resource if it exists
-	if isOSAuthEnabled {
-		reqLogger.Info("OSAUTH_ENABLED set to true, look for matching OAuthClient resource to delete")
-		clientId := client.Spec.ClientId
-		err = r.deleteOAuthClient(ctx, clientId)
-		if err != nil {
-			return
-		}
-	}
+	reqLogger.Info("OSAUTH_ENABLED set to true, deleting annotations from ibm-iam-operand-restricted ServiceAccount")
```
[the remark this answers is not in the document]
Comment (Tirumalavasa) [reply pkg/controller/client/client_controller.go]: addressed it

File: pkg/controller/authentication/sacc.go [hunk 2]
```diff
@@ -111,3 +119,30 @@ func generateSAObject(instance *operatorv1alpha1.Authentication, scheme *runtime
 	}
 	return operandSA
 }
+
+func (r *ReconcileAuthentication) getConsoleURL(instance *operatorv1alpha1.Authentication) string {
```
[the remark this answers is not in the document]
Comment (Tirumalavasa) [reply pkg/controller/authentication/sacc.go:124]: addressed

**Comment by Tirumalavasa** [comment]
> @Tirumalavasa or @rashmi43 - would you please also fixup the commits that git log noise prior to merging, too? I'd like for our git log to be more readable, and having bullets for things like initial commits and merges in this context gets in the way of what the commit accomplished (removed OAuthClient support, specific behavior in SA's around annotations.)

@rwhundley,Right. I always squash all the commits into once before i ensure Final PR is made, This time i forgot to do it, as there were couple of pulls i did from release-future.

Whoever merges this PR, Lets do `Squash and Merge` and keep single commit msg in the list. This way our commit history on release-future will be clean and meaningful

File: pkg/controller/client/client_controller.go [hunk 3]
```diff
@@ -406,18 +397,8 @@ func (r *ReconcileClient) updateClient(ctx context.Context, client *oidcv1.Clien
...
-		return err
-	}
-	if isOSAuthEnabled {
-		reqLogger.Info("OSAUTH_ENABLED set to true, look for matching OAuthClient resource to potentially update")
-		_, err = r.reconcileOAuthClient(ctx, client)
-		if err != nil {
-			return err
-		}
-	}
+	reqLogger.Info("OSAUTH_ENABLED set to true, updating annotations to ibm-iam-operand-restricted ServiceAccount")
```
[the remark this answers is not in the document]
Comment (rashmi43) [reply pkg/controller/client/client_controller.go]: it will be removed in the next PR when we support IDP registration for OS

**Review by rwhundley — APPROVED** [review]

**Pull request merged**